docs/client-libraries-schema.md

```diff
@@ -142,4 +142,96 @@ while True:
     except Exception:
         # Message failed to be processed
         consumer.negative_acknowledge(msg)
-```
\ No newline at end of file
+```
+
+## Work with the Go schema
+Working with Go schema has slight differences from Java schema.
+This part will introduce the schema compatibility between Go client and Java client.
+
+#### Avro Schema
+Avro schema in Go and Java are compatible, but there are some differences in how the schemas are defined. Go typically uses schema definitions, a string JSON, to create schemas. However, Java often uses class types for schema creation. As a result, Java allows non-primitive fields to be nullable by default, while in Go's schema definition, the nullability of fields needs to be explicitly stated.
+GO:
+```go
+// Compatible with defining a schema in Java
+exampleSchemaDefCompatible := NewAvroSchema(`{"fields":
+    [
+        {"name":"id","type":"int"},{"default":null,"name":"name","type":["null","string"]}
+    ],
+    "name":"MyAvro","namespace":"schemaNotFoundTestCase","type":"record"}`, nil)
+// Not compatible with defining a schema in Java
```

Comment:
Does this refer to the following Java example?

If the Go Client is used like this, the Java client should also support it, right? 

In java client, there are two ways:
1. Use SchemaDefinition to set schema JsonDef

``` 
          String jsonDef = "{"fields":[{"name":"id","type":"int"},{"default":null,"name":"name","type":["null","string"]}],"name":"MyAvro","namespace":"schemaNotFoundTestCase","type":"record"}"
           Schema schema = Schema.AVRO(SchemaDefinition.builder().withJsonDef("").build());
```

2. Or, to set not allow field is null.
```
      SchemaDefinition<Example> schemaDefinition =
                     SchemaDefinition.builder().withPojo(Example.class).withAlwaysAllowNull(false).build();
      Schema schema = Schema.AVRO(schemaDefinition);
      
    Producer<Example> producer = pulsarClient.newProducer(schema)
                .topic(topic).create();
```

we should show the solution.